docs/advanced_topics/customisation/page_editing_interface.rst

```diff
@@ -57,22 +57,99 @@ Wagtail provides a general-purpose WYSIWYG editor for creating rich text content
 
 However, template output from :class:`~wagtail.wagtailcore.fields.RichTextField` is special and need to be filtered to preserve embedded content. See :ref:`rich-text-filter`.
 
-If you're interested in extending the capabilities of the Wagtail WYSIWYG editor (``hallo.js``), See :ref:`extending_wysiwyg`.
+
+.. _rich_text_features:
+
+Limiting features in a rich text field
+--------------------------------------
+
+By default, the rich text editor provides users with a wide variety of options for text formatting and inserting embedded content such as images. However, we may wish to restrict a rich text field to a more limited set of features - for example:
+
+ * The field might be intended for a short text snippet, such as a summary to be pulled out on index pages, where embedded images or videos would be inappropriate;
+ * When page content is defined using :ref:`StreamField <streamfield>`, elements such as headings, images and videos are usually given their own block types, alongside a rich text block type used for ordinary paragraph text; in this case, allowing headings and images to also exist within the rich text content is redundant (and liable to result in inconsistent designs).
+
+This can be achieved by passing a ``features`` keyword argument to ``RichTextField``, with a list of identifiers for the features you wish to allow:
+
+.. code-block:: python
+
+    body = RichTextField(features=['p', 'h2', 'h3', 'b', 'i', 'link'])
+
+The recognised feature identifiers are as follows (note that add-on modules may add to this list):
```

Comment:
I'd prefer to leave this out until we see that there's a really strong real-world need for it... I'm wary of making the features system "too powerful". The issue we ran into on #2175, where we passed form widget objects as parameters and then found that they broke migrations, was a hint to me that we shouldn't really be stuffing detailed configuration options about the editor's behaviour into model fields.

My intention with features - which will probably become clearer once they start being used in the whitelisting / text conversion logic - is that they're describing the data format ("this field is allowed to contain `<h2>` headings") rather than the behaviour of the editor, and then it's the editor widget's responsibility to decide how to make that available to the user (most likely by adding an 'h2' button to the toolbar).

The example of disallowing nested `<ul>`s is an interesting one, because you can make a strong case that that _is_ a detail of the data format, not just the editor's behaviour. My worry is that developers won't grasp this distinction, and will end up misusing per-feature configuration options to control editor behaviour ("I want to add a button with icon _foo_, which opens a chooser at URL _bar_...") far more often than using them legitimately.